### PR TITLE
Add checks in Date methods

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #195.

### CODE REVIEW

File: glib/src/date.rs [hunk 1]
```diff
@@ -130,73 +152,125 @@ impl Date {
...
+                ffi::g_date_subtract_days(self.to_glib_none_mut().0, n_days);
+            }
+            Ok(())
         }
     }
 
-    pub fn subtract_months(&mut self, n_months: u32) {
+    pub fn subtract_months(&mut self, n_months: u32) -> Result<(), BoolError> {
+        // The checks for this function are just a mess in the C code, allowing intermediate
+        // unknonwn state. So for now, nothing can be done...
```
Comment (sdroege) [inline glib/src/date.rs:259]:
```suggestion
        // unknown state. So for now, nothing can be done...
```

`subtract_days()` should be much worse then, but you don't have such a comment there :)

**Comment by sdroege** [comment]
I guess this is better than nothing...

File: glib/src/date.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply glib/src/date.rs:259]: Hum no, `subtract_days` is easy to check in comparison: https://gitlab.gnome.org/GNOME/glib/-/blob/master/glib/gdate.c#L1709

Comment (sdroege) [reply glib/src/date.rs]: Ah because it goes through julian days...

**Review by sdroege — APPROVED** [review]

**Pull request merged**